botorch/optim/optimize_mixed.py

```diff
@@ -123,15 +125,26 @@ def _filter_infeasible(
             `\sum_i (X[indices[i]] * coefficients[i]) >= rhs`. `indices` and
             `coefficients` should be torch tensors. See the docstring of
             `make_scipy_linear_constraints` for an example.
+        equality_constraints: A list of tuples (indices, coefficients, rhs),
+            with each tuple encoding an equality constraint of the form
+            `\sum_i (X[indices[i]] * coefficients[i]) == rhs`. `indices` and
+            `coefficients` should be torch tensors. See the docstring of
+            `make_scipy_linear_constraints` for an example.
 
     Returns:
         The tensor `X` with infeasible points removed.
     """
-    if inequality_constraints is None:
+    if inequality_constraints is None and equality_constraints is None:
         return X
     is_feasible = torch.ones(X.shape[:-1], device=X.device, dtype=torch.bool)
-    for idx, coef, rhs in inequality_constraints:
-        is_feasible &= (X[..., idx] * coef).sum(dim=-1) >= rhs
+    if inequality_constraints is not None:
+        for idx, coef, rhs in inequality_constraints:
+            is_feasible &= (X[..., idx] * coef).sum(dim=-1) >= rhs
+    if equality_constraints is not None:
+        for idx, coef, rhs in equality_constraints:
+            is_feasible &= torch.isclose(
+                (X[..., idx] * coef).sum(dim=-1), torch.tensor(rhs).to(coef)
+            )
```

Comment:
Could you use `evaluate_feasibility` from `botorch.optim.parameter_constraints` to increase code-sharing? If we need to change a tolerance, it would be better to not have to do that in multiple places.

Reply:
Good point, will do so!

Reply:
I had a deeper look into `evaluate_feasibility` https://github.com/pytorch/botorch/blob/17a066bf7bba95400925a826212c52bf93ab1ad2/botorch/optim/parameter_constraints.py#L613

It is not directly applicable here as it always checks feasibility on the q-batch level and not on the row level as needed here. The reason why the checks are perfomed on the q level is that it also checks for interpoint constraints. One could modify the behavior of the method by introducing a flag if it should also check for inter point constraints, and then perform the check on the row level.

What do you think?